presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -129,13 +143,48 @@ public void testJoinDynamicFilteringSingleValue()
         assertEquals(result.getResult().getRowCount(), 6);
 
         // Probe-side is dynamically filtered:
-        QueryStats stats = runner.getCoordinator().getQueryManager().getFullQueryInfo(result.getQueryId()).getQueryStats();
-        Set<Long> rowsRead = stats.getOperatorSummaries()
+        Set<Long> rowsRead = getOperatorRowsRead(runner, result.getQueryId());
+        assertEquals(rowsRead, ImmutableSet.of(6L, ORDERS_COUNT));
+    }
+
+    @Test
+    public void testJoinDynamicFilteringBlockProbeSide()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(ENABLE_DYNAMIC_FILTERING, "true")
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, FeaturesConfig.JoinReorderingStrategy.NONE.name())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.BROADCAST.name())
+                .build();
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        // Wait for both build sides to finish before starting the scan of 'lineitem' table (should be very selective given the dynamic filters).
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                session,
+                "SELECT l.comment" +
+                        " FROM  lineitem l, part p, orders o" +
+                        " WHERE l.orderkey = o.orderkey AND o.comment = 'nstructions sleep furiously among '" +
+                        " AND p.partkey = l.partkey AND p.comment = 'onic deposits'");
+        assertEquals(result.getResult().getRowCount(), 1);
+        assertEquals(getOperatorRowsRead(runner, result.getQueryId()), ImmutableSet.of(1L, ORDERS_COUNT, PART_COUNT));
+
+        // Make sure that a single join doesn't doesn't narrow lineitem to single row.
```

Comment:
you can remove this test case. It's essentially identical to the one above

Reply:
Oops, thanks for catching this!
Remove the duplicate and added a new test case to `testJoinDynamicFilteringSingleValue`.